### PR TITLE
calculate currentSubchainStart when loading from local storage

### PR DESCRIPTION
**Pull request by oconnor663**
Previously we only did it during server load, which worked in tests but
not when the current user was loaded from cache after a service restart.
Factor out some of the subchain start logic so that we can share it with
LoadLinksFromStorage.

r? @maxtaco 

cc @zapu

### CODE REVIEW

File: go/libkb/sig_chain.go [hunk 1]
```diff
@@ -363,53 +364,85 @@ var hardcodedResets = map[keybase1.SigID]bool{
...
+	// Confirm that the last link is not stubbed. This would prevent us from
+	// reading the eldest_kid, so the server should never do it.
+	lastLink := sc.chainLinks[len(sc.chainLinks)-1]
+	if lastLink.IsStubbed() {
+		return nil, errors.New("the last chain link is unexpectedly stubbed in GetCurrentSunchain")
+	}
+	// Check whether the eldest KID doesn't match the latest link. That means
+	// the account has just been reset, and so as with a new account, there is
+	// no current subchain.
+	if !sc.chainLinks[len(sc.chainLinks)-1].ToEldestKID().Equal(eldest) {
```
Comment (maxtaco) [inline go/libkb/sig_chain.go]: How did we get away with not doing this before?

Comment (maxtaco) [inline go/libkb/sig_chain.go]: (Seems like a good idea, but wasn't it done somewhere else in the code?)

File: go/libkb/sig_chain.go [hunk 2]
```diff
@@ -705,61 +745,58 @@ func (l *SigChainLoader) AccessPreload() bool {
...
 	var mt *MerkleTriple
-	goodKey := true
 
 	uid := l.user.GetUID()
 
 	l.G().Log.CDebugf(l.ctx, "+ SigChainLoader.LoadFromStorage(%s)", uid)
 	defer func() { l.G().Log.CDebugf(l.ctx, "- SigChainLoader.LoadFromStorage(%s) -> %s", uid, ErrToOk(err)) }()
 
-	if mt, err = l.LoadLastLinkIDFromStorage(); err != nil || mt == nil {
+	if mt, err = l.LoadLastLinkIDFromStorage(); err != nil || mt == nil || mt.LinkID == nil {
```
Comment (maxtaco) [inline go/libkb/sig_chain.go]: What case is this and why did it work previously?

File: go/libkb/sig_chain.go (hunk 1 above)
Comment (oconnor663) [inline go/libkb/sig_chain.go]: This was on line 413 before. In the current version, `GetCurrentSubchain` is only responsible for checking the very last link against the expected eldest KID. Eldest KID changes prior to that in the chain are left to `isSubchainStart`, which is shared with cache loading, and which doesn't need an eldest KID from the Merkle tree.

File: go/libkb/sig_chain.go (hunk 2 above)
Comment (oconnor663) [inline go/libkb/sig_chain.go]: I kind of can't remember. But part of what's changed is that we used to do basically this check on line 761, and now this is the only place that does it.

**Pull request merged**